### ISSUE TITLE
Enabling CORS generates route conflict message

### ISSUE THREAD

**Issue by aulvi** [hapijs/hapi#2927]
Using the code below, when I attempt to start the server, I get the following error:

"Error: New route /hello conflicts with existing /hello/{foo?}"

"Working" example can be found here: https://github.com/aulvi/hapi-help.

Node 4.x, NPM 3.x, Hapi 10.5.0

Not sure what I'm doing wrong, any advice is welcomed, thanks!

```
var Hapi = require('hapi');

var server = new Hapi.Server();
server.connection({ port: 3000 });

server.route({
  config: {
    cors: true
  },
  method: 'GET',
  path: '/hello/{foo?}',
  handler: function (request, reply) {
    reply('Hello, world!\n');
  }
});

server.route({
  config: {
    cors: true
  },
  method: 'POST',
  path: '/hello',
  handler: function (request, reply) {
    reply('Hello, world!\n');
  }
});

server.start(function () {
  console.log('Server running at:', server.info.uri);
});
```

**Comment by hueniverse** [comment]
I can't reproduce the error with the code above.

**Comment by bboysathish** [comment]
@aulvi R u trying like this?

```js
var Hapi = require('hapi');

var server = new Hapi.Server();
server.connection({ port: 3000 });

server.route({
  config: {
    cors: true
  },
  method: 'POST',
  path: '/hello/{foo?}',
  handler: function (request, reply) {
    reply('Hello, world!\n');
  }
});

server.route({
  config: {
    cors: true
  },
  method: 'GET',
  path: '/hello/{foo?}',
  handler: function (request, reply) {
    reply('Hello, world!\n');
  }
});

server.route({
  config: {
    cors: true
  },
  method: 'POST',
  path: '/hello',
  handler: function (request, reply) {
    reply('Hello, world!\n');
  }
});

server.start(function () {
  console.log('Server running at:', server.info.uri);
});
```

Then you will get an error,

```js
Error: New route /hello conflicts with existing /hello/{foo?}
    at Object.exports.assert (d:\E-Learning\hapi-killer\node_modules\hapi\node_modules\hoek\lib\index.js:731:11)
    at internals.Segment.add (d:\E-Learning\hapi-killer\node_modules\hapi\node_modules\call\lib\segment.js:49:14)
    at internals.Router.add (d:\E-Learning\hapi-killer\node_modules\hapi\node_modules\call\lib\index.js:62:26)
    at internals.Connection._init.internals.Connection._start.internals.Connection._stop.internals.Connection._dispatch.internals.Conne
    at internals.Connection._init.internals.Connection._start.internals.Connection._stop.internals.Connection._dispatch.internals.Conne
    at module.exports.internals.Plugin.internals.Plugin.register.internals.Plugin._apply (d:\E-Learning\hapi-killer\node_modules\hapi\l
    at module.exports.internals.Plugin.internals.Plugin.register.internals.Plugin.route (d:\E-Learning\hapi-killer\node_modules\hapi\li
    at Object.<anonymous> (d:\E-Learning\hapi-killer\sample.js:28:8)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
```

**Comment by hueniverse** [comment]
That'e because you can't have a route with optional param and one that's the same without the param. That's a clear routing conflict.

**Comment by aulvi** [comment]
@hueniverse The methods are different, one path is a POST and the other a GET. If I remove `cors: true` then the server starts and works as expected.

If this is a clear routing conflict, why is the error not present if cors is not enabled?

Thanks!

**Comment by bboysathish** [comment]
@aulvi Your test case works perfectly with removing `cors` and setting `cors`.

**Comment by aulvi** [comment]
@bboysathish If you clone my repository, `npm install` and `node index.js` does it work for you? I have tried this on two different systems (osx and linux) and both behave the same way.

```
git clone https://github.com/aulvi/hapi-help
cd hapi-help
npm install
node index.js
```

Then I get an error.

What am I missing? Thanks!

**Comment by aulvi** [comment]
@bboysathish I cannot reproduce this error with Hapi 11.x, only with the 10.x line. Is it possible you are testing my example with a newer version of Hapi?

**Comment by bboysathish** [comment]
@aulvi  Apologies, i tested it with 11.0.x, now i am able to reproduce that error. 

1. With `cors` error is thrown.
2. By removing `cors` on anyone of the route, not throws error. 

`Hapi@ 10.5.0`

**Comment by hueniverse** [comment]
This makes sense now. Yes, this was fixed in 11.0.0.
